File: Python/Easy/Palindrome Number.py
# https://leetcode.com/problems/palindrome-number/

class Solution(object):
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
        x_copy = x
        reverse = 0

        while x_copy > 0:
            reverse += x_copy % 10
            x_copy /= 10
            reverse *= 10
                    
        reverse /= 10

        return reverse == x

File: C++/Easy/Palindrome Number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        long int orig = x;
        long int reversed = 0;
        
        while(x > 0) {
            reversed += x % 10;
            x /= 10;    
            reversed *= 10;
        }
        reversed /= 10;

        // cout << reversed << endl;
        // cout << orig << endl;
        // cout << (reversed == orig) << endl;
        
        return reversed == orig;
    }
};
